v03_pipeline/lib/misc/pedigree.py

```diff
@@ -107,56 +124,34 @@ def parse_collateral_lineage(
         # A sample_i that is siblings with sample_j, will list sample_j as as sibling, but
         # sample_j will not list sample_i as a sibling.  Relationships only appear in the
         # ibd table a single time, so we only need to check the pairing once.
-        for sample_i, sample_j in itertools.combinations(samples.keys(), 2):
-            # If other sample is already related, continue
-            if sample_j in {
-                samples[sample_i].mother,
-                samples[sample_i].father,
-                samples[sample_i].maternal_grandmother,
-                samples[sample_i].maternal_grandfather,
-                samples[sample_i].paternal_grandmother,
-                samples[sample_i].paternal_grandfather,
-            }:
+        for sample_i, sample_j in itertools.combinations(samples.values(), 2):
+            # If sample is already related from direct relationships, continue
+            if sample_i.is_in_direct_lineage(sample_j):
```

Comment:
quick summary - what's the logic of what it was doing before versus after?

Reply:
the bug was in how we were checking for sample being in a direct lineage.  We were only checking a single direction (sample_i is mother/father/grandparent of sample_j) when we need to check both.  

The below logic which checks for "aunt-uncle" makes the assumption that samples aren't direct related, so that check needs to actually be correct.